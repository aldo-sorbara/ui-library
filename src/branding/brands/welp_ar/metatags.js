export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
