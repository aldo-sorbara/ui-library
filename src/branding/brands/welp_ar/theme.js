import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
