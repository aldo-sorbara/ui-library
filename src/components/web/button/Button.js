import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * @render react
 * @name Button
 * @description Button Component for Wenance.
 * @example
 * <ButtonContainer>
 *   <Button data="button"> Button</Button>
 * </ButtonContainer>
 */

export const Button = props => {
  const { children, data, href } = props;
  const Component = href ? StyledLink : StyledButton;
  return (
    <Component {...props} data-w={data}>
      {children}
    </Component>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  data: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  isBlocked: PropTypes.bool,
  modifiers: PropTypes.arrayOf(PropTypes.string)
};

Button.defaultProps = {
  disabled: false,
  isBlocked: false,
  modifiers: []
};

const CommonButton = props => `
  border-radius: ${props.theme.buttons.borderRadius};
  display: ${props.isBlocked ? 'block' : 'inline-block'};
  cursor: pointer;
  font-family: inherit;
  opacity: ${props.disabled ? '0.3' : '1'};
  padding: ${props.theme.spacer.medium} ${props.theme.spacer.wrapperInner};
  text-align: center;
  transition: background-color 0.4s ease-in-out;
  touch-action: manipulation;
  user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: ${props.isBlocked ? '100%' : 'auto'};
  cursor: ${props.disabled ? 'not-allowed' : 'pointer'};
  background-color: ${props.theme.palette.primary.default};
  border: 1px solid transparent;
  color: ${props.theme.palette.common.text.inverse};

  &:hover:enabled {
    background-color: ${props.theme.palette.primary.dark};
  }

  &:focus {
    outline: none;
  }

  &:disabled,
  button[disabled] {
    opacity: 0.3;
  }
`;

CommonButton.defaultProps = {
  theme: {
    palette: {
      common: {
        text: {
          inverse: '#ffffff'
        }
      },
      primary: {
        default: '#0082ff',
        dark: '#3a66ab'
      }
    },
    typography: {
      button: {
        fontSize: '.875rem',
        fontWeight: '700',
        lineHeight: '1.145'
      }
    },
    spacer: {
      medium: '1rem',
      large: '1.5rem',
      button: {
        smallSpacer: '.75rem'
      }
    },
    buttons: {
      borderRadius: '0'
    }
  }
};

const StyledButton = styled.button`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
`;

const StyledLinkButton = props => `
  background: transparent;
  border: 1px solid transparent;
  color: ${props.theme.palette.primary.default};
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }`;

const StyledLink = styled.a`
  ${CommonButton}
  ${props => props.theme.typography.button};
  ${props => props.modifiers.map(modifier => modifiers[modifier])}
  text-decoration: none;
`;

const StyledSmallButton = props => `
  padding: ${props.theme.spacer.button.smallSpacer} ${props.theme.spacer.large};
`;

const StyledInverseButton = props => `
  background: transparent;
  color: ${props.theme.palette.common.text.inverse};
  border: 1px solid ${props.theme.palette.common.text.inverse};
  
  &:hover, &:hover:enabled {
    background-color: inherit;
    opacity: .65
  }
`;

const StyledRoundedButton = () => `
  border-radius: 50%;
  width: 50px;
  height: 50px
  padding: 0;

  svg {
    display: block;
    margin: auto;
  }
`;

const modifiers = {
  small: StyledSmallButton,
  rounded: StyledRoundedButton,
  link: StyledLinkButton,
  inverse: StyledInverseButton
};

// ButtonContainer Component
export const ButtonContainer = props => {
  return (
    <StyledButtonContainer {...props}>
      <div>{props.children}</div>
    </StyledButtonContainer>
  );
};

const StyledButtonContainer = styled.div`
  padding: ${props => props.theme.buttons.padding};
`;

StyledButtonContainer.defaultProps = {
  theme: {
    buttons: {
      padding: '0 2rem 2rem'
    }
  }
};
