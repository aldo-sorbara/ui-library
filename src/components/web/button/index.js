export { Button, ButtonContainer } from './Button';
