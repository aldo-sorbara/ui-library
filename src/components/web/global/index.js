export {
  Container,
  Wrapper,
  Headline,
  Description,
  Legales,
  Label,
  LabelLarge,
  SecondHeadline,
  ContainerDesktop,
  Desktop,
  Onboarding,
  Terms,
  H1,
  SectionDescription,
  UrlLink,
  Timer,
  DesktopText,
  Bind,
  Logo,
  LogoMobile,
  TextPromotion
} from './Global';
