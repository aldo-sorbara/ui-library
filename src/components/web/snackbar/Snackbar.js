import React from 'react';
import PropTypes from 'prop-types';
import styled, { withTheme, keyframes } from 'styled-components';

const IconInfo = () => (
  <svg
    xmlns='http://www.w3.org/2000/svg'
    width='24'
    height='24'
    viewBox='0 0 24 24'
  >
    <path fill='none' d='M0 0h24v24H0z' />
    <path
      fill='currentColor'
      d='M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z'
    />
  </svg>
);

const snackbarShow = keyframes`
      0%{ bottom:-70px; }
      10% {bottom:15px; }
      95% {bottom:15px; }
      100% {bottom:-80px}
`;

const Snackbar = props => {
  const { content, title, Icon, action, modifier, fixed } = props;
  const dataw = props[`data-w`];

  return (
    <SnackbarContainer data-w={dataw} fixed={fixed} modifier={modifier}>
      <Box>
        <IconContainer content={content}>{Icon || <IconInfo />}</IconContainer>
        <p>
          <span>
            <strong>{title}</strong>
          </span>
          {content && <span>{content}</span>}
        </p>
      </Box>
      {action && <ActionContainer>{action}</ActionContainer>}
    </SnackbarContainer>
  );
};

Snackbar.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string,
  fixed: PropTypes.bool,
  modifier: PropTypes.oneOf(['info', 'success', 'error', 'warning']),
  action: PropTypes.node
};
Snackbar.defaultProps = {
  modifier: 'info',
  content: null,
  fixed: false
};

const Box = styled.div`
  align-items: center;
  display: flex;
  p {
    margin-bottom: 0;
  }
  span:nth-child(2) {
    display: block;
    margin-top: 0.5rem;
  }
`;
const IconContainer = styled.span`
  background-color: ${props => props.theme.palette.common.bodyBg};
  border-radius: 50%;
  display: inline-block;
  flex-shrink: 0;
  font-size: 0;
  height: 40px;
  justify-content: center;
  margin-right: 0.75rem;
  ${props => props.content && { alignSelf: 'center', bottom: '2px' }};
  position: relative;
  text-align: center;
  width: 40px;
  & > svg {
    position: relative;
    top: 0.5rem;
  }
`;

IconContainer.defaultProps = {
  theme: {
    palette: {
      common: {
        bodyBg: '#FFF'
      }
    }
  }
};

const ActionContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin-left: 0.75rem;
`;

const SnackbarContainer = styled.div`
  animation: ${snackbarShow} 4s ease-in-out;
  background-color: ${props => props.theme.palette[props.modifier].light};
  border: 1px solid transparent;
  color: ${props => props.theme.palette[props.modifier].dark};
  display: flex;
  justify-content: space-between;
  line-height: 1.25;
  margin-top: ${props =>
    props.theme.spacer.blockSpacerFb || props.theme.spacer.blockSpacer};
  ${props => props.theme.alerts.container}
  ${props =>
    props.fixed && {
      position: 'fixed',
      left: 0,
      right: 0,
      alignItems: 'center',
      margin: '1rem'
    }}
  padding: 0.75rem 1rem 0.75rem 1rem;

  &:empty {
    display: none;
  }

  & ~ & {
    margin-top: ${props => props.theme.spacer.xSmall};
  }

  & > p {
    margin-top: 0;

    &:last-child {
      margin-bottom: 0;
    }
  }
  z-index: 9999;
`;

SnackbarContainer.defaultProps = {
  theme: {
    spacer: {
      blockSpacer: '1.25rem',
      xSmall: '.5rem'
    }
  }
};

export default withTheme(Snackbar);
