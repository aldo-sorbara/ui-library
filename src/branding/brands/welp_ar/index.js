import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
