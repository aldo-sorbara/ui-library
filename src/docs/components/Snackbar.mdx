---
name: Snackbar
menu: Components
route: /snackbar
---

import { Playground, PropsTable } from 'docz';
import Snackbar from '../../components/web/snackbar';
import { Button } from '../../components/web/button';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';
import { styles } from '../../../utils/docz';

# Card

<PropsTable of={Snackbar} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
       <Snackbar
        title={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured}
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        action={
          <Button modifiers={['link', 'rounded']} data='dismiss'>
            <svg
              xmlns='http://www.w3.org/2000/svg'
              fill='currentColor'
              height='24'
              width='24'
              viewBox='0 0 35 35'
            >
              {/* eslint-disable max-len */}
                <path d='M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z' />
              {/* eslint-enable max-len */}
            </svg>
          </Button>
        }
      />
       <Snackbar
        title={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured}
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        modifier="success"
      />
       <Snackbar
        title={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured}
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        modifier="warning"
      />
       <Snackbar
        title={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured}
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        modifier="error"
      />
    </>

  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/snackbar--with-default-props)
