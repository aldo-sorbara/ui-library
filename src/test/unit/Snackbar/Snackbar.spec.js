import React from 'react';
import Snackbar from '../../../components/web/snackbar';
import { Button } from '../../../components/web/button';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Snackbar component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Snackbar
          title={
            welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured
          }
          content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a success snackbar', () => {
    const wrapper = mountWithTheme(
      <Snackbar
        title={
          welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured
        }
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        modifier='success'
      />,
      welp_ar.theme
    );
    expect(wrapper.props().modifier).toEqual('success');
  });
  it('should render a warning snackbar', () => {
    const wrapper = mountWithTheme(
      <Snackbar
        title={
          welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured
        }
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        modifier='warning'
      />,
      welp_ar.theme
    );
    expect(wrapper.props().modifier).toEqual('warning');
  });
  it('should render an error snackbar', () => {
    const wrapper = mountWithTheme(
      <Snackbar
        title={
          welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured
        }
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        modifier='error'
      />,
      welp_ar.theme
    );
    expect(wrapper.props().modifier).toEqual('error');
  });
  it('should render a snackbar with action', () => {
    const wrapper = mountWithTheme(
      <Snackbar
        title={
          welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured
        }
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        action={<Button data='dismiss'>Dismiss</Button>}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().action).not.toBe(undefined);
  });
  it('should render a fixed snackbar', () => {
    const wrapper = mountWithTheme(
      <Snackbar
        title={
          welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured
        }
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
        action={<Button data='dismiss'>Dismiss</Button>}
        fixed
      />,
      welp_ar.theme
    );
    expect(wrapper.props().fixed).toBe(true);
    expect(wrapper).toHaveStyleRule('position', 'fixed');
  });
  it('should render a snackbar without spacerFb', () => {
    const theme = welp_ar.theme;
    theme.spacer.blockSpacerFb = null;
    const wrapper = mountWithTheme(
      <Snackbar
        title={
          welp_ar.wording.onboarding.basicInfo.inputSMS.alert.contentFeatured
        }
        content={welp_ar.wording.onboarding.basicInfo.inputSMS.alert.content}
      />,
      theme
    );
    expect(wrapper).toHaveStyleRule('margin-top', '1.25rem');
  });
});
