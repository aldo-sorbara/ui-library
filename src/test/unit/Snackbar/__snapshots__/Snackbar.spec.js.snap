// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar component should render snapshot 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 p {
  margin-bottom: 0;
}

.c2 span:nth-child(2) {
  display: block;
  margin-top: 0.5rem;
}

.c3 {
  background-color: #ffffff;
  border-radius: 50%;
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-size: 0;
  height: 40px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 0.75rem;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  bottom: 2px;
  position: relative;
  text-align: center;
  width: 40px;
}

.c3 > svg {
  position: relative;
  top: 0.5rem;
}

.c1 {
  -webkit-animation: bNHMNQ 4s ease-in-out;
  animation: bNHMNQ 4s ease-in-out;
  background-color: #e0f6f9;
  border: 1px solid transparent;
  color: #127a87;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 1.25;
  margin-top: 0;
  font-size: .875rem;
  padding: 0.75rem 1rem 0.75rem 1rem;
  z-index: 9999;
}

.c1:empty {
  display: none;
}

.c1 ~ .c0 {
  margin-top: .25rem;
}

.c1 > p {
  margin-top: 0;
}

.c1 > p:last-child {
  margin-bottom: 0;
}

<div
  className="c0 c1"
>
  <div
    className="c2"
  >
    <span
      className="c3"
      content="Si no recibiste el código podés reenviarlo usando el siguiente link."
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
          fill="currentColor"
        />
      </svg>
    </span>
    <p>
      <span>
        <strong>
          ¡Revisá tus mensajes!
        </strong>
      </span>
      <span>
        Si no recibiste el código podés reenviarlo usando el siguiente link.
      </span>
    </p>
  </div>
</div>
`;
